File: curso-python/ex056.py

# leia o nome, idade e sexo de 4 pessoas. e no final do programa, mostre
somaIdade = 0
idade_homem_velho = 0
nome_homem_velho = ''
totMulher20 = 0

for pessoa in range(1,5):
    print("{} {}°a PESSOA {}".format('-'*5,pessoa,'-'*5))
    nome = input("Nome: ").strip()
    idade = int(input("idade: "))
    sexo = input("Sexo [M/F]: ").strip().upper()

    # somar as médias
    somaIdade += idade

    #verificando o homem mais velho do grupo
    if sexo in 'Mm' and idade > idade_homem_velho:
        idade_homem_velho = idade
        nome_homem_velho = nome
        
    #verificando mulheres com menos de 20 anos
    if sexo in "Ff" and idade < 20:
        totMulher20 += 1
        

# vou pegar a soma das médias e dividir por 4
media = somaIdade / 4
#a média de idade do grupo de pessoas
print(f"a média de idade do grupo é de {media}")
#o homem mais velho
print(f"o homem mais velho é {nome_homem_velho} e sua idade é {idade_homem_velho} anos")
#quantas mulheres com menos de 20 anos
print(f"Ao todo são {totMulher20} mulheres com menos de 20 anos")

File: javascript-exercicios/ex056.js
const prompt = require("prompt-sync")();

let somaIdade = 0;
let idade_homem_velho = 0;
let nome_homem_velho = '';
let totMulher20 = 0;

for(let pessoa=0; pessoa < 5; pessoa++){
    console.log(`------------ ${pessoa}°a PESSOA ---------------`);
    let nome = prompt("Nome: ").trim();
    let idade = parseInt(prompt("idade: "));
    let sexo = prompt("Sexo [M/F]: ").trim().toUpperCase()

    somaIdade += idade

    if ((sexo === "M") && (idade > idade_homem_velho)){
        idade_homem_velho = idade;
        nome_homem_velho = nome;
    }
    if((sexo === "F") && (idade < 20)){
        totMulher20 += 1
    }
}

let media = somaIdade / 4;

console.log(`\nA média de idade do grupo é de ${media.toFixed(1)} anos`);
if (nome_homem_velho) {
    console.log(`O homem mais velho é ${nome_homem_velho} e tem ${idade_homem_velho} anos`);
} else {
    console.log(`Não foi informado nenhum homem no grupo`);
}
console.log(`Ao todo são ${totMulher20} mulher(es) com menos de 20 anos`);